src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathMapperTest.java

```diff
@@ -0,0 +1,533 @@
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.internal.TestUtils;
+import net.sf.saxon.om.AxisInfo;
+import net.sf.saxon.om.Item;
+import net.sf.saxon.sxpath.XPathDynamicContext;
+import net.sf.saxon.sxpath.XPathEvaluator;
+import net.sf.saxon.sxpath.XPathExpression;
+import net.sf.saxon.trans.XPathException;
+import net.sf.saxon.tree.iter.EmptyIterator;
+
+public class XpathMapperTest {
+
+    @Test
+    public void testFullPath() throws Exception {
+        final String xpath = "/CLASS_DEF/OBJBLOCK/METHOD_DEF[1]/SLIST/VARIABLE_DEF[2]";
+        final DetailAST rootAst = getRootAst("InputXpathMapperAst.java");
+        final RootNode rootNode = new RootNode(rootAst);
+        final DetailAST[] actual = convertToDetailAstArray(getXpathItems(xpath, rootNode));
+        final DetailAST realVariableDefNode = getSiblingByType(rootAst, TokenTypes.CLASS_DEF)
+                .findFirstToken(TokenTypes.OBJBLOCK)
+                .findFirstToken(TokenTypes.METHOD_DEF)
+                .findFirstToken(TokenTypes.SLIST)
+                .findFirstToken(TokenTypes.VARIABLE_DEF)
+                .getNextSibling()
+                .getNextSibling();
+        final DetailAST[] expected = {realVariableDefNode};
+        assertArrayEquals(expected, actual);
+    }
+
+    @Test
+    public void testParent() throws Exception {
+        final String xpath = "(//VARIABLE_DEF)[1]/..";
+        final DetailAST rootAst = getRootAst("InputXpathMapperAst.java");
+        final RootNode rootNode = new RootNode(rootAst);
+        final DetailAST[] actual = convertToDetailAstArray(getXpathItems(xpath, rootNode));
```

Comment:
I think better and more clear name is `convertToArray` by arguments and return type it is clear that it about DetailAST